throughs/flatten.js

```diff
@@ -3,27 +3,33 @@
 var values = require('../sources/values')
 var once = require('../sources/once')
 
+function getAbortCb(read, abort, cb) {
+  return function(err) {
+    read(err || abort, cb)
+  }
+}
+function getChunkErrorCb(err, cb) {
+  return function() {
+    cb(err)
+  }
+}
+
 //convert a stream of arrays or streams into just a stream.
 module.exports = function flatten () {
   return function (read) {
     var _read
     return function (abort, cb) {
       if (abort) { //abort the current stream, and then stream of streams.
-        _read ? _read(abort, function(err) {
-          read(err || abort, cb)
-        }) : read(abort, cb)
+        _read ? _read(abort, getAbortCb(read, abort, cb)) : read(abort, cb)
```

Comment:
I would ask you to revert this change, because it creates some indirection in the order of reading code, and since we have so many cases of nested closure functions around the tech stack, we could also have to apply this optimization everywhere else and that would end up disrupting code readability quite a lot. I don't see why this optimization is a big gain, I'm more interested in the algorithmic changes that usually give order of magnitude gains. It could also be that this code optimization has different performance numbers across different JS engines (V8, Spidermonkey, Hermies) or even across different versions of the same JS engine (V8 has changed considerably, people have adapted their code style to be V8-performance friendly and they have had to keep changing it over the years).

Reply:
Reverted.